### DISCUSSION TITLE
Interested in contributing to Alaska projects for GSoC 2026 (AI/ML)

### DISCUSSION THREAD

**Discussion by lakshyadchad** [uaanchorage/GSoC#45] [General]
Hey @pradeeban sir and mentors,
I'm Lakshya Sharma from Indian a CSE major (AI & ML) student with experience in:
Real life project with ML Frameworks (pytorch and tensorflow) to Build AI models.
Python and SQL for backend development 
I am interested in contributing to the Automated coastline extraction for erosion modeling in Alaska project for GSoC 2026.
I have reviewed the project description and I look forward to learning from the community and contributing.

Thank you!
Lakshya Sharma

**Comment by lakshyadchad** [comment]
Hi @pradeeban sir and @fwitmer mentor,

"What I have did right now is set up a Conda environment to run this project locally and wrote a simulation script to replicate the NDWI logic.

My Initial Results: I successfully generated a water mask where the Deering airport runway is correctly classified as land. However, I noticed that the 'No Data' padding around the rotated image is currently being misclassified as water using a simple thresholding approach. I plan to implement masking for these 'No Data' regions next.

You can view my reproduction script here: https://github.com/lakshyadchad/alaska-gsoc-prep

I am particularly interested in the "Cliff Area Segmentation" challenge mentioned in the project scope. are there any specific documentation updates or small reproduction tasks that would be helpful to the team? pls give suggestion and replay for improvement

<img width="1718" height="572" alt="Screenshot 2025-12-30 005626" src="https://github.com/user-attachments/assets/7be9ddf0-caf7-4ec5-bf67-ed1190277c15" />
<img width="1920" height="1080" alt="Screenshot 2025-12-30 012324" src="https://github.com/user-attachments/assets/b67c5df0-32af-49f3-b85b-2b193401f46f" />
<img width="1640" height="887" alt="Screenshot 2025-12-30 130931" src="https://github.com/user-attachments/assets/f506f791-61ce-4ab8-905a-0355f38b022c" />

Looking forward to contributing!

**Comment by lakshyadchad** [comment]
@pradeeban 
Please let me know if there is any other media, like Slack or Discord, related to this project for discussion, because no one is replying or saying anything on the discussion page of https://github.com/fwitmer/CoastlineExtraction.

**Reply by pradeeban** [reply]
Hi @lakshyadchad I see this is discussed successfully in that project repository. Pls note your messages were during the Christmas-New Year vacation and we likely missed it.

**Comment by lakshyadchad** [comment]
Progress Update: Vectorization Pipeline & Border Artifact Fix

I have been working on the extraction pipeline. I successfully implemented a script that fix No data problem that "white border" artifacts (where nodata was previously being misclassified as water)
by adding the Feature: vectorizes the binary mask into Shapely polygons.

You can view my vector script here: https://github.com/lakshyadchad/alaska-gsoc-prep

<img width="1920" height="1080" alt="Screenshot 2025-12-30 143027" src="https://github.com/user-attachments/assets/74a04ef6-4cd1-4fc4-9e5e-36d450d11ba4" />

thank you
